### PR TITLE
feat: Update AmazonBedrockChatGenerator to use Converse API (BREAKING CHANGE)

### PR DESCRIPTION
**Pull request by vblagoje**
Enhances the integration with Amazon Bedrock by using the new Converse API in `AmazonBedrockChatGenerator`. The modifications address the removal of outdated adapter structures and simplifies code greatly all while maintaining compatibility as much as possible.

### What:
- **Removed the adapter file**: The `adapters.py` file was deleted to eliminate deprecated structures related to chat model adapters.
- **Updated `chat_generator.py`**: 
  - Improved the model selection process and model ID usage.
  - Revised how system prompts and messages are prepared for sending to the models.
  - Added support for streaming responses and tool configurations in model requests.
- **Enhanced tests**: Significant updates in `test_chat_generator.py` to include new model tests and validate functionality with the new configurations and expected responses.

### How can it be used:
- The system now supports:
  - Multiple LLMs from Amazon Bedrock with configurations for model-specific parameters.
  - Streaming responses during chat interactions.
- Example code snippet to run a chat generator:
    ```python
    from haystack_integrations.components.generators.amazon_bedrock import AmazonBedrockChatGenerator

    client = AmazonBedrockChatGenerator(model="anthropic.claude-3-5-sonnet-20240620-v1:0")
    response = client.run(
        messages=[ChatMessage.from_user("What's Natural Language Processing?")],
        generation_kwargs={"maxTokens": 512}
    )
    ```

### How did you test it:
Conducted integration tests that focused on:
- Ensuring the correct integration of the `AmazonBedrockChatGenerator` with various model configurations.
- Validating the model's ability to handle both standard and streaming requests.
- Mocked AWS session interactions to simulate real response cycles using the updated chat models.

### Notes for the reviewer:
- Attention should be given to changes affecting model naming conventions and interaction patterns, especially regarding tool usage in the LLM context.
- Review the removal of the `adapters.py` file to understand how changes in architecture affect existing functionalities.

### Breaking Changes
- The `truncate` initialization parameter has been removed from `AmazonBedrockChatGenerator`

#### Migration Steps:
- Remove any `truncate` parameter from component initialization.

### CODE REVIEW

**Comment by vblagoje** [comment]
@julian-risch @silvanocerza do you remember who was giving us permissions for Amazon Bedrock models, was it @askainet ? We need to ping him to click through all these Amazon Bedrock models in this PR so they become available on our Haystack account...

File: integrations/amazon_bedrock/tests/test_chat_generator.py
Comment (vblagoje) [inline integrations/amazon_bedrock/tests/test_chat_generator.py:23]: These models in MODELS_TO_TEST, MODELS_TO_TEST_WITH_TOOLS and STREAMING_TOOL_MODELS @askainet 🙏 🙏

**Comment by vblagoje** [comment]
@anakin87 would you please give it a rough pass first to make sure I'm not forgetting any major parts.

**Review by anakin87 — COMMENTED** [review]
I had a cursory look and I like the change and the approach, also because we are deleting a lot of code :-)

Ping me when you need a more in-depth review.

File: integrations/amazon_bedrock/tests/test_chat_generator.py
Comment (vblagoje) [reply integrations/amazon_bedrock/tests/test_chat_generator.py:23]: Paul set us up 🚀

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py:230]: we can move `try` closer to the actual client usage
(after `callback = streaming_callback or self.streaming_callback`)

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py:259]: I think we can easily add an unit test for this method

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py:291]: I would also like to see a unit test for this method, but IDK how tricky it is, so I'll leave it up to you

**Review by anakin87 — CHANGES_REQUESTED** [review]
I am happy with this refactoring. (Just left some minor comments)

One relevant aspect I noticed involves truncation - we removed it and I am OK with it but we should:
- remove the `truncate` parameter
- clearly communicate this in the PR title (which is used for changelog generation)
- release this breaking change in a major version once this PR has been merged

**Comment by vblagoje** [comment]
Addressed your concerns @anakin87 
Not 100% sure about:
```
from botocore.eventstream import EventStream
```

direct usage to improve clarity but since we already import botocore I though it was ok - look like stable API. LMK your thoughts here and please have another look!

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py:222]:
```suggestion
            system_prompts = [{"text": messages[0].text}]
```

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py:225]:
```suggestion
        messages_list = [{"role": msg.role.value, "content": [{"text": msg.text}]} for msg in messages]
```

**Review by anakin87 — APPROVED** [review]
LGTM!

Please incorporate the suggested changes to align with #1236: in short, `text` has been introduced to smooth transition to new `ChatMessage` in future but currently it just mirrors `content`.

Once this PR is merged, please release 2.0.0

**Pull request merged**
